components/server/src/server.ts

```diff
@@ -85,7 +85,10 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
             res.send = (body) => {
                 const result = originalSend(body);
                 const method = req.method;
-                const route = req.route.path;
+                const route = req.route?.path || "";
```

Comment:
@akosyakov, why using `req.route` and not `req.path` here? is `prometheus-metrics` expecting the relative routes, or would full path' work as well? 

@corneliusludmann what is the effect of passing empty values in `prometheus-metrics`?

Reply:
because we are not interested in concrete values, but how much some route is hit

Reply:
Just to clarify:
- route path is `/v1/resource/:resource/:ref`
- url path is instantiation of such route, i.e. `/v1/resource/machines/xyz`

We are interested in former, not latter.

Discussed with Alex that we keep it like that, since it could be interested which URLs are hit often but don't have any route matcher.

Reply:
> what is the effect of passing empty values in `prometheus-metrics`?

Hey @ArthurSens! Could you help out with your expertise? `route` could be an empty string here:

https://github.com/gitpod-io/gitpod/blob/e2bf23a914c8115ec6665a1602a3835c0f66b80d/components/server/src/server.ts#L90-L95

Which results in an empty label here:

https://github.com/gitpod-io/gitpod/blob/e2bf23a914c8115ec6665a1602a3835c0f66b80d/components/server/src/prometheus-metrics.ts#L77-L77

and here: 

https://github.com/gitpod-io/gitpod/blob/e2bf23a914c8115ec6665a1602a3835c0f66b80d/components/server/src/prometheus-metrics.ts#L89-L89

(`route` is an empty string, other labels have values).

Could this lead to problems in production?

---

```typescript
/**
* Increment for given labels
* @param labels Object with label keys and values
* @param value The number to increment with
* @param timestamp Timestamp to associate the time series with
*/
inc(labels: labelValues, value?: number, timestamp?: number | Date): void;
```

```typescript
interface labelValues {
	[key: string]: string | number;
}
```

Reply:
btw we should be able to check it in staging metrics already

Reply:
> We are interested in former, not latter.

We could URL path as fallback to get a better picture of missing routes. Don't know if it's confusing in the metrics, though.

Reply:
I thinking it is enough to have them in logs. It can be expensive if happens often, because of high cardinality.